### PR TITLE
Add Info::Applied

### PR DESCRIPTION
**Pull request by FintanH**
When having tracked a peer we don't immediately have their changes. We
can know that we do have the peer's changes when we have fetched their
changes or if we have applied them from a gossip message.

We need a way to surface that a result was successfully applied. In this
patch we do this by adding it to the `Info` enum and emit the protocol
message to the subscribers.

### CODE REVIEW

File: seed/src/lib.rs [hunk 1]
```diff
@@ -251,6 +251,15 @@ impl Node {
...
             },
+            ProtocolEvent::Gossip(gossip::Info::Applied(peer::Gossip { urn, rev, origin })) => {
+                let peer_id = origin.map(|peer| peer.to_string());
+                tracing::info!(
+                    "Applied change for URN `{}`, rev = {:?}, peer.id = {:?}",
+                    urn,
+                    rev,
+                    peer_id
+                );
+            },
```
Comment (cloudhead) [inline seed/src/lib.rs:262]: This is gonna be super nice to have!

File: librad/src/net/gossip.rs [hunk 2]
```diff
@@ -101,6 +101,7 @@ where
     Addr: Clone + PartialEq + Eq + Hash,
 {
     Has(Has<Addr, A>),
+    Applied(A),
```
Comment (cloudhead) [inline librad/src/net/gossip.rs:104]: Does this belong here? Shouldn't this only be for messages that are actually gossiped between peers?

Comment (FintanH) [reply librad/src/net/gossip.rs:104]: I'm not sure I understand the question, could you elaborate? :)

Comment (cloudhead) [reply librad/src/net/gossip.rs:104]: Should the `Applied` variant be here, since it won't be broadcast to peers?

Comment (FintanH) [reply librad/src/net/gossip.rs:104]: `Info` isn't for broadcasting to peers. It's used for announcing protocol events, afaict.

Comment (FintanH) [reply librad/src/net/gossip.rs:104]: Where protocol events are internal for the current peer

**Comment by kim** [comment]
Technically, this is not really _surfacing_ anything -- would you not want to be notified via the `PeerEvent`s?

**Comment by FintanH** [comment]
> Technically, this is not really surfacing anything -- would you not want to be notified via the PeerEvents?

Ah, well it seems I'm confused between the two :sweat_smile: It seems like this may not actually be needed because I surfaced the result in my previous set of changes: https://github.com/radicle-dev/radicle-link/blob/maint/librad/src/net/peer.rs#L124. Would that be right?

**Pull request closed without merging**